### PR TITLE
Fix handling dates earlier than 1900 in Oracle backend

### PR DESCRIPTION
**Pull request by JaroslawCendrowski**
Solves https://github.com/SOCI/soci/issues/1135

### CODE REVIEW

**Comment by vadz** [comment]
I think this solves the problem for the years between 1800 and 1900, but not those earlier than 1800, right? This is probably better than nothing, but it looks like we could easily solve it for any negative `tm_year` in the same way.

But beyond this I also wonder if this actually works in the other backends. It would be great to add a unit test checking for this to the "std::tm" section of the corresponding test in `tests/common-tests.h`.

**Comment by JaroslawCendrowski** [comment]
It fixes years in starting from 1. Didn't test for years BC yet.

> I think this solves the problem for the years between 1800 and 1900, but not those earlier than 1800, right? This is probably better than nothing, but it looks like we could easily solve it for any negative `tm_year` in the same way.
> 
> But beyond this I also wonder if this actually works in the other backends. It would be great to add a unit test checking for this to the "std::tm" section of the corresponding test in `tests/common-tests.h`.

**Comment by JaroslawCendrowski** [comment]
Test added. It fails with current implementation, works with my changes in standard-use-type.cpp.

> I think this solves the problem for the years between 1800 and 1900, but not those earlier than 1800, right? This is probably better than nothing, but it looks like we could easily solve it for any negative `tm_year` in the same way.
> 
> But beyond this I also wonder if this actually works in the other backends. It would be great to add a unit test checking for this to the "std::tm" section of the corresponding test in `tests/common-tests.h`.

File: tests/oracle/test-oracle.cpp [hunk 1]
```diff
@@ -85,6 +93,51 @@ TEST_CASE("Oracle datetime", "[oracle][datetime]")
...
+        table_creator_for_timestamp tableCreator(sql);
+
+        for(int i = 1; i <= 2201; i = i + 50)
+        {
+            char t[10];
+            sprintf(t, "%04d", i);
+
+            std::string date = std::to_string(i) + "-03-28 14:06:13";
+            std::tm t1 {}, t2 {}, t4 {};
+            strptime(date.c_str(), "%Y-%m-%d %H:%M:%S", &t2);
```
Comment (vadz) [inline tests/oracle/test-oracle.cpp:110]: I don't think we can use `strptime()` under Windows, unfortunately.

Comment (JaroslawCendrowski) [reply tests/oracle/test-oracle.cpp]: Changed to std::get_time, unfortunately it doesn't work well with years < 100, so I changed the tested years range also.
Tested on Windows and Ubuntu.

**Comment by vadz** [comment]
Thanks, I'm (squash) merging this now!

P.S. Oracle "excess-100 notation" is really weird, I wonder who came up with something like this and why.

**Pull request closed without merging**